### PR TITLE
feat: Initial REST API endpoints for assignments.

### PR DESCRIPTION
**Pull request by pwnage101**
ENT-7680

### CODE REVIEW

File: enterprise_access/apps/api/v1/views/content_assignments/assignments.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    viewsets.GenericViewSet,
+):
+    """
+    Viewset supporting all Admin CRUD operations on ``LearnerContentAssignment`` records.
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = serializers.LearnerContentAssignmentResponseSerializer
+    authentication_classes = (JwtAuthentication, authentication.SessionAuthentication)
+    filter_backends = (filters.NoFilterOnDetailBackend,)
+    filterset_class = filters.AssignmentFilter
```
Comment (iloveagent57) [inline enterprise_access/apps/api/v1/views/content_assignments/assignments.py:33]: Do we need filtering on the learner version?  And if so, can learners filter on the same fields as admins?

**Review by iloveagent57 — COMMENTED** [review]
This is great, one question about list-filtering for learners.

File: enterprise_access/apps/api/v1/views/content_assignments/assignments.py (hunk 1 above)
Comment (pwnage101) [reply enterprise_access/apps/api/v1/views/content_assignments/assignments.py:33]: Right, at least not the same as admins.  The one's the learners might conceivably every want to filter on are content_key and state.  I don't know if the learner portal would ever need to do that filtering though.

Comment (pwnage101) [reply enterprise_access/apps/api/v1/views/content_assignments/assignments.py:33]: Maybe we just not pre-emtively introduce dead code and not allow learners to filter the list results.

Comment (iloveagent57) [reply enterprise_access/apps/api/v1/views/content_assignments/assignments.py:33]: Either one sounds reasonable to me!

Comment (iloveagent57) [reply enterprise_access/apps/api/v1/views/content_assignments/assignments.py:33]: anyway, this is my only concern, I'll approve so you can merge at your convenience

**Review by iloveagent57 — APPROVED** [review]

**Pull request merged**